gulpfile.js

```diff
@@ -108,15 +109,17 @@ function getAssetGroups() {
             assetGroups.push(assetGroup);
         });
     });
+
     return assetGroups;
 }
 
 function resolveAssetGroupPaths(assetGroup, assetManifestPath) {
     assetGroup.manifestPath = assetManifestPath;
     assetGroup.basePath = path.dirname(assetManifestPath);
-    assetGroup.inputPaths = assetGroup.inputs.map(function (inputPath) {
+    const inputPaths = assetGroup.inputs.map(function (inputPath) {
         return path.resolve(path.join(assetGroup.basePath, inputPath)).replace(/\\/g, '/');
     });
+    assetGroup.inputPaths = inputPaths.sort();
```

Comment:
Since this is already about hard-coded input paths (right?) we needn't sort them. Or rather, we mustn't sort them, since the manually specified order might matter in the resulting bundle.

Reply:
I actually don't think this sort is needed because we sort later after grabbing the files gulp.src. If you want, remove that sort but keep the other and see if we still get the same results. If that does not work, you can re-add it.

Reply:
It works without sorting at all in the gulpfile too (see https://github.com/OrchardCMS/OrchardCore/pull/15772#discussion_r1567914546).

Reply:
I don't think it will work without some sort of sort. Because when we use path that contains a star, some tasks may resolve slower than other which cause the concatenation to not be consistent. When you force the sort, this ensures that the files are always concatenated in the same order. This leads to generating the same file over time.

Reply:
Yes, and I mean, this is why we need manual sorting, but in the Assets files. The gulpfile needn't know about that.

This isn't an ideal solution (the whole assets pipeline is far from ideal), but without fundamentally re

Reply:
I suggest using the gulp-sort for now. Honestly, the entire gulp setup should be replaced with another solution at some point. These is probably other dependencies that we already use that have been also marked absolute and is not maintainable.

For now, gulp-sort will continue to do the job until we revamp the assets setup.

Reply:
There are three cases of a wildcard lookup in OC that needed the paths manually listed. Until we add any new files in those three folders, we're done. And if we do add new files there, it's about adding that path to the Assets file (i.e. the same as in the other cases where there's no wildcard). I think we can live with that, and I'd prefer this as an interim solution rather than relying on a library that was last updated two major Gulp versions prior.

And the Gulp pipeline is already riddled with issues:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/334a5693-5c71-4278-bf1d-8c84b860c08c)

Reply:
These 3 places are where it was used and the order matters, ever (since this only affects the OC source). Why do we need anything more complex before we fix this for good and delete this anyway?

If we were to keep this process for the foreseeable future then I wouldn't be happy with this either. But I don't see the point in trying to patch a fundamentally outdated system to maybe prevent a 5-second inconvenience in the future if somebody adds new files.

Reply:
@Piedone look at the latest comment in this PR. I fixed the issue without having to depend on any new library. 

change the `resolveAssetGroupPaths` function to this

```
function resolveAssetGroupPaths(assetGroup, assetManifestPath) {
    assetGroup.manifestPath = assetManifestPath.replace(/\\/g, '/');
    assetGroup.basePath = path.dirname(assetGroup.manifestPath);
    var inputPaths = assetGroup.inputs.map(function (inputPath) {
        return path.resolve(path.join(assetGroup.basePath, inputPath)).replace(/\\/g, '/');
    });

    // get all input paths and then sort them to ensure file concatenation is consistent.
    assetGroup.inputPaths = glob.sync(inputPaths, {}).sort();

    assetGroup.watchPaths = [];
    if (!!assetGroup.watch) {
        assetGroup.watchPaths = assetGroup.watch.map(function (watchPath) {
            return path.resolve(path.join(assetGroup.basePath, watchPath)).replace(/\\/g, '/');
        });
    }
    assetGroup.outputPath = path.resolve(path.join(assetGroup.basePath, assetGroup.output)).replace(/\\/g, '/');
    assetGroup.outputDir = path.dirname(assetGroup.outputPath);
    assetGroup.outputFileName = path.basename(assetGroup.output);
    // Uncomment to copy assets to wwwroot
    //assetGroup.webroot = path.join("./src/OrchardCore.Cms.Web/wwwroot/", path.basename(assetGroup.basePath), path.dirname(assetGroup.output));
}
```

update `glob` to "glob": "^10.3.12",

Reply:
I'll move this to your PR.